Fix some performance-related warnings reported by cppcheck

cdumez opened the pull request:
#### 4d53eb48f4c606624a2887a6e094e6eb1f2d640a
<pre>
Fix some performance-related warnings reported by cppcheck
<a href="https://bugs.webkit.org/show_bug.cgi?id=254854">https://bugs.webkit.org/show_bug.cgi?id=254854</a>

Reviewed by Darin Adler.

* Source/WebCore/Modules/fetch/FetchBody.h:
(WebCore::FetchBody::FetchBody):
* Source/WebCore/Modules/websockets/WebSocketDeflater.cpp:
(WebCore::WebSocketInflater::WebSocketInflater):
* Source/WebCore/css/MediaQueryParserContext.cpp:
(WebCore::MediaQueryParserContext::MediaQueryParserContext):
* Source/WebCore/css/parser/CSSParserObserverWrapper.cpp:
(WebCore::CSSParserObserverWrapper::skipCommentsBefore):
(WebCore::CSSParserObserverWrapper::yieldCommentsBefore):
* Source/WebCore/css/parser/CSSParserSelector.cpp:
(WebCore::CSSParserSelector::CSSParserSelector):
* Source/WebCore/display/css/DisplayStyle.cpp:
(WebCore::Display::Style::Style):
* Source/WebCore/inspector/InspectorStyleSheet.h:
(WebCore::InspectorCSSId::InspectorCSSId):
* Source/WebCore/loader/LinkPreloadResourceClients.cpp:
(WebCore::LinkPreloadResourceClient::LinkPreloadResourceClient):
* Source/WebCore/page/UserContentURLPattern.cpp:
(WebCore::UserContentURLPattern::UserContentURLPattern):
* Source/WebCore/page/ViewportConfiguration.cpp:
(WebCore::ViewportConfiguration::ViewportConfiguration):
* Source/WebCore/platform/DragImage.cpp:
* Source/WebCore/platform/ThreadSafeDataBuffer.h:
(WebCore::ThreadSafeDataBufferImpl::ThreadSafeDataBufferImpl):
(WebCore::ThreadSafeDataBuffer::ThreadSafeDataBuffer):
* Source/WebCore/platform/graphics/Color.h:
(WebCore::Color::OutOfLineComponents::create):
(WebCore::Color::OutOfLineComponents::OutOfLineComponents):
* Source/WebCore/platform/image-decoders/png/PNGImageDecoder.cpp:
(WebCore::PNGImageReader::PNGImageReader):
* Source/WebCore/platform/ios/SelectionGeometry.cpp:
(WebCore::operator&lt;&lt;):
* Source/WebCore/platform/ios/SelectionGeometry.h:
* Source/WebCore/style/StyleScopeRuleSets.cpp:
(WebCore::Style::ScopeRuleSets::ScopeRuleSets):
* Source/WebCore/svg/SVGPathByteStreamSource.cpp:
(WebCore::SVGPathByteStreamSource::SVGPathByteStreamSource):
* Source/WebCore/testing/MallocStatistics.h:
(WebCore::MallocStatistics::MallocStatistics):
* Source/WebKit/Platform/IPC/unix/UnixMessage.h:
(IPC::UnixMessage::UnixMessage):

Canonical link: <a href="https://commits.webkit.org/262468@main">https://commits.webkit.org/262468@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ba4d8e3d89f2c5d67ea88ff64b1a1c5e640031aa

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/1596 "2 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/1627 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/1684 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/2517 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/1717 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/1575 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/1692 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/1689 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/1500 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/1610 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/1449 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/1446 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/2354 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/1448 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/1432 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/1348 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/1454 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/1472 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/2523 "run-api-tests-without-change (failure)") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/1490 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/1334 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/1406 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/1439 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/398 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/1553 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/css/MediaQueryParserContext.cpp:
```diff
@@ -33,14 +33,14 @@
 }
 
 MediaQueryParserContext::MediaQueryParserContext(const Document& document)
+    : useSystemAppearance(document.page() ? document.page()->useSystemAppearance() : false)
```
I would use `&&` here.

darinadler commented on Source/WebCore/platform/DragImage.cpp:
```diff
@@ -256,9 +256,9 @@ DragImage::DragImage(DragImageRef dragImageRef)
 DragImage::DragImage(DragImage&& other)
     : m_dragImageRef { std::exchange(other.m_dragImageRef, nullptr) }
+    , m_indicatorData { other.m_indicatorData }
+    , m_visiblePath { other.m_visiblePath }
```
These should have `WTFMove` or `std::exchange`.

cdumez committed:
Fix some performance-related warnings reported by cppcheck
https://bugs.webkit.org/show_bug.cgi?id=254854

Reviewed by Darin Adler.

* Source/WebCore/Modules/fetch/FetchBody.h:
(WebCore::FetchBody::FetchBody):
* Source/WebCore/Modules/websockets/WebSocketDeflater.cpp:
(WebCore::WebSocketInflater::WebSocketInflater):
* Source/WebCore/css/MediaQueryParserContext.cpp:
(WebCore::MediaQueryParserContext::MediaQueryParserContext):
* Source/WebCore/css/parser/CSSParserObserverWrapper.cpp:
(WebCore::CSSParserObserverWrapper::skipCommentsBefore):
(WebCore::CSSParserObserverWrapper::yieldCommentsBefore):
* Source/WebCore/css/parser/CSSParserSelector.cpp:
(WebCore::CSSParserSelector::CSSParserSelector):
* Source/WebCore/display/css/DisplayStyle.cpp:
(WebCore::Display::Style::Style):
* Source/WebCore/inspector/InspectorStyleSheet.h:
(WebCore::InspectorCSSId::InspectorCSSId):
* Source/WebCore/loader/LinkPreloadResourceClients.cpp:
(WebCore::LinkPreloadResourceClient::LinkPreloadResourceClient):
* Source/WebCore/page/UserContentURLPattern.cpp:
(WebCore::UserContentURLPattern::UserContentURLPattern):
* Source/WebCore/page/ViewportConfiguration.cpp:
(WebCore::ViewportConfiguration::ViewportConfiguration):
* Source/WebCore/platform/DragImage.cpp:
* Source/WebCore/platform/ThreadSafeDataBuffer.h:
(WebCore::ThreadSafeDataBufferImpl::ThreadSafeDataBufferImpl):
(WebCore::ThreadSafeDataBuffer::ThreadSafeDataBuffer):
* Source/WebCore/platform/graphics/Color.h:
(WebCore::Color::OutOfLineComponents::create):
(WebCore::Color::OutOfLineComponents::OutOfLineComponents):
* Source/WebCore/platform/image-decoders/png/PNGImageDecoder.cpp:
(WebCore::PNGImageReader::PNGImageReader):
* Source/WebCore/platform/ios/SelectionGeometry.cpp:
(WebCore::operator<<):
* Source/WebCore/platform/ios/SelectionGeometry.h:
* Source/WebCore/style/StyleScopeRuleSets.cpp:
(WebCore::Style::ScopeRuleSets::ScopeRuleSets):
* Source/WebCore/svg/SVGPathByteStreamSource.cpp:
(WebCore::SVGPathByteStreamSource::SVGPathByteStreamSource):
* Source/WebCore/testing/MallocStatistics.h:
(WebCore::MallocStatistics::MallocStatistics):
* Source/WebKit/Platform/IPC/unix/UnixMessage.h:
(IPC::UnixMessage::UnixMessage):

Canonical link: https://commits.webkit.org/262468@main
